### ISSUE TITLE
[QUESTION] Why is the namespace hardcoded in the prometheus addon (and maybe others)?

### ISSUE THREAD

**Issue by spkane** [aws-ia/terraform-aws-eks-blueprints#498]
#### Please describe your quesiton here

The namespace for the prometheus addon is hardcoded. Why is this? 

I have spent a bit of time trying to change this, before realizing why I couldn't get it to change as expected.

#### Provide link to the example related to the question

https://github.com/aws-ia/terraform-aws-eks-blueprints/blob/deec7d5caea47c06dea48fa616ad2c56e52c3cce/modules/kubernetes-addons/prometheus/locals.tf#L8

#### Additional context

Similar issue/fix: https://github.com/aws-ia/terraform-aws-eks-blueprints/pull/287

#### More

- [X ] Yes, I have checked the repo for existing issues before raising this question

**Comment by bryantbiggs** [comment]
I can't answer the why but I think instead we should default to what the associated addon's helm chart provides while offering the users the ability to change to suite. Changing this from a question to an enhancement

**Comment by spkane** [comment]
Agreed. I just wasn't sure if there was a reason for this, so I figured I'd start by asking.

**Comment by vara-bonthu** [comment]
@spkane What you are seeing in `locals.tf` are the default values for each Add-on.  All the Add-ons can be configured by the users to pass their own values. 

Did you try passing your own config with `prometheus_helm_config={}` ? 

Here is the doc https://aws-ia.github.io/terraform-aws-eks-blueprints/add-ons/prometheus/

**Comment by spkane** [comment]
@vara-bonthu I had this setup in my code when I was testing it:

```hcl
  enable_prometheus                   = true
  prometheus_helm_config = {
    version          = "15.8.5"
    create_namespace = true
    namespace        = "monitoring"
    values           = [templatefile("${path.module}/helm_values/prometheus-values.yaml.tftpl", {nodeSelector = local.primaryNodeSelector})]
  }
```

but since it has been a week, let me test it one more time and double check.

**Comment by spkane** [comment]
So, it looks like the monitoring namespace is created, but that the ArgoCD application/helm is not targeting the right namespace.

![CleanShot 2022-05-13 at 08 27 07](https://user-images.githubusercontent.com/129629/168316764-288e66e4-2955-4510-b237-33988eb7a684.png)

Digging in just a bit more...

**Comment by spkane** [comment]
Ahhh. I think the error might actually be here:

https://github.com/aws-samples/eks-blueprints-add-ons/blob/main/chart/templates/prometheus.yaml#L33


It looks like the destination namespace is hard-coded here (and maybe in other examples) and does not respect what was set in the Terraform code.
